specs/CookieManagement.md

```diff
@@ -0,0 +1,158 @@
+    _cookieManager.DeleteCookies("CookieName", "https://www.bing.com", "", "");
+}
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    /// Kind of cookie SameSite status used in the CoreWebView2Cookie class.
+    /// These fields match those as specified in https://developer.mozilla.org/docs/Web/HTTP/Cookies#.
+    /// Learn more about SameSite cookies here: https://tools.ietf.org/html/draft-west-first-party-cookies-07
+    enum CoreWebView2CookieSameSiteKind
+    {
+        /// None SameSite type. No restrictions on cross-site requests.
+        None = 0,
+        /// Lax SameSite type. The cookie will be sent with "same-site" requests, and with "cross-site" top level navigation.
+        Lax = 1,
+        /// Strict SameSite type. The cookie will only be sent along with "same-site" requests.
+        Strict = 2,
+    };
+
+    runtimeclass CoreWebView2
+    {
+        /// Create a new cookie manager object.
+        CoreWebView2CookieManager CreateCookieManager();
+    }
+
+    /// Add or delete an CoreWebView2Cookie or CoreWebView2Cookies,
+    /// or view the cookies. The changes would apply to the context of the user profile.
+    /// That is, other WebViews under the same user profile could be affected.
+    runtimeclass CoreWebView2CookieManager
+    {
+        /// Create a cookie object with a specified name, value, domain, and path.
+        /// One can set other optional properties after cookie creation.
+        CoreWebView2Cookie CreateCookie(String name, String value, String Domain, String Path);
+
+        /// Gets a list of cookies matching the specific URI.
+        /// You can modify the cookie objects, call
+        /// CoreWebView2CookieManager.SetCookie, and the changes
+        /// will be applied to the webview.
+        Windows.Foundation.IAsyncOperation<CoreWebView2CookieList> GetCookiesAsync(String uri);
+
+        /// Sets a cookie with the given cookie data; may overwrite equivalent cookies
+        /// if they exist.
+        void SetCookie(CoreWebView2Cookie cookie);
+
+        /// Deletes browser cookies with matching name and uri or domain/path pair.
+        /// Cookie name is required.
+        /// If uri is specified, deletes all cookies with the given name where domain
+        /// and path match provided URI.
+        /// If domain is specified, deletes only cookies with the exact domain.
+        /// If path is specified, deletes only cookies with the exact path.
+        void DeleteCookies(String name, String uri, String Domain, String Path);
+
+        /// Clears all cookies under the same profile.
+        /// This could affect other WebViews under the same user profile.
+        void ClearAllCookies();
+    }
+
+    /// A list of cookie objects.
+    runtimeclass CoreWebView2CookieList
+    {
+        /// The number of cookies contained in the CoreWebView2CookieList.
+        UInt32 Size { get; };

```

Comment:
For .NET users I think that it should be called `Length` or `Count` rather than Size. If you take a look at existing .NET collection types (i.e. List, Dictionary, etc.pp) they are all either called `Length` or `Count` but never `Size`.

Reply:
In .NET `CoreWebView2CookieList` should implement `System.Collections.Generic.IList<CoreWebView2Cookie>` and there would be a `Count` property.